packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -251,11 +252,13 @@ class ReactSixteenAdapter extends EnzymeAdapter {
         } else {
           isDOM = false;
           const { type: Component } = el;
+
           const isStateful = Component.prototype && (
             Component.prototype.isReactComponent
             || Array.isArray(Component.__reactAutoBindPairs) // fallback for createClass components
           );
-          if (!isStateful) {
+
+          if (!isStateful && typeof Component === 'function') {
```

Comment:
This is just so we get a clearer error is a user tries to mount a Fragment or Portal, they will see the ShallowTestRenderer  informative error instead of the much less clear "Component is not a function"